test/integration/index.test.js

```diff
@@ -45,48 +47,93 @@ const testResponseXlsx = (res) => {
     expect(res.headers['content-type']).toEqual('application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
 };
 
-test('able to create server', () => {
-    createMockServer();
-});
+describe('report', () => {
+    let config;
 
-test('able to create server without any reports configured', () => {
-    createMockServer({ templatesDir: path.join(__dirname, 'emptyDir') });
-});
+    beforeEach(() => {
+        config = JSON.parse(JSON.stringify(defaultConfig));
+    });
 
-test('healthcheck passes', async () => {
-    const res = await createMockServer().get('/');
-    expect(res.statusCode).toEqual(200);
-    expect(res.body).toStrictEqual({ status: 'ok' });
-    testResponse(res);
-});
+    it('able to create server', () => {
+        createMockServer({ config });
+    });
 
-test('default mock report - CSV - correct response', async () => {
-    const res = await createMockServer().get('/test.csv?currency=MMK');
-    expect(res.statusCode).toEqual(200);
-    expect(csvParse(res.text, { columns: true })).toStrictEqual([
-        {
-            Currency: 'MMK',
-            Name: 'fsp1',
-        },
-        {
-            Currency: 'MMK',
-            Name: 'fsp3',
-        },
-    ]);
-    testResponse(res, { contentType: 'csv' });
-});
+    test('healthcheck passes', async () => {
+        const res = await createMockServer({ config }).get('/');
+        expect(res.statusCode).toEqual(200);
+        expect(res.body).toStrictEqual({ status: 'ok' });
+        testResponse(res);
+    });
 
-test('default mock report - XLSX - correct response', async () => {
-    const res = await createMockServer().get('/test.xlsx?currency=MMK');
-    expect(res.statusCode).toEqual(200);
-    expect(res.body).toStrictEqual({});
-    testResponseXlsx(res, { contentType: 'xlsx' });
-});
+    test('CSV - correct response', async () => {
+        const server = createMockServer({ config });
+        const watch = k8s.Watch.getInstance();
+        watch.sendResource(path.join(__dirname, 'data/test.yaml'));
+        const res = await server.get('/test?dfspId=payerfsp&currency=MMK&format=csv');
+        expect(res.statusCode).toEqual(200);
+        expect(csvParse(res.text, { columns: true })).toStrictEqual([
+            {
+                Currency: 'MMK',
+                Name: 'fsp1',
+            },
+            {
+                Currency: 'MMK',
+                Name: 'fsp3',
+            },
+        ]);
+        testResponse(res, { contentType: 'csv' });
+    });
+
+    test('XLSX - correct response', async () => {
+        const server = createMockServer({ config });
+        const watch = k8s.Watch.getInstance();
+        watch.sendResource(path.join(__dirname, 'data/test.yaml'));
+        const res = await server.get('/test?dfspId=payerfsp&currency=MMK&format=xlsx');
+        expect(res.statusCode).toEqual(200);
+        expect(res.body).toStrictEqual({});
```

Comment:
This is surprising to me. Why does the result not have a body?

Reply:
The file content is in the `res.text` field